### PR TITLE
ticdc: Update ticdc mysql sink description

### PR DESCRIPTION
**Pull request by 7yyo**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Update ticdc `mysql sink` description.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

Update ticdc `mysql sink` description.

Please confirm whether the serial write change is added in `v4.0.5`.

https://github.com/pingcap/ticdc/pull/843/files 

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-overview.md [hunk 1]
```diff
@@ -54,7 +54,8 @@ TiCDC 的系统架构如下图所示：
 
 - MySQL sink
 
-    - TiCDC 不拆分表内事务，**保证**单表事务一致性，但**不保证**与上游表内事务的顺序一致。如果事务间的主键有交集，依然可以保证与上游顺序一致。
+    - TiCDC 不拆分单表事务，**保证**单表事务原子性。
+    - TiCDC **保证**单表事务顺序一致性，**注意** v4.0.5 之前版本没有主键或唯一索引的表**不保证**顺序一致性。
```
Comment (amyangfei) [inline ticdc/ticdc-overview.md:58]: https://github.com/pingcap/ticdc/pull/843 doesn't change the result of causality detection, so the delivery guarantee in MySQL sink is not changed.
Have you met unexpected event sequence in test?

Comment (7yyo) [reply ticdc/ticdc-overview.md:58]: I will modify this description later

File: ticdc/ticdc-overview.md [hunk 2]
```diff
@@ -54,7 +54,8 @@ TiCDC 的系统架构如下图所示：
...
     - TiCDC 以表为单位拆分跨表事务，**不保证**跨表的事务始终一致。
```
Comment (leoppro) [inline ticdc/ticdc-overview.md:59]:
```suggestion
    - TiCDC 以表为单位拆分跨表事务，**不保证**跨表的事务原子性。
```

File: ticdc/ticdc-overview.md (hunk 1 above)
Comment (leoppro) [inline ticdc/ticdc-overview.md:58]:
```suggestion
    - TiCDC **不保证**下游事务执行顺序和上游完全一致。
```

**Comment by 7yyo** [comment]
@leoppro @amyangfei PTAL

**Comment by leoppro** [comment]
LGTM

File: ticdc/ticdc-overview.md [hunk 3]
```diff
@@ -54,8 +54,9 @@ TiCDC 的系统架构如下图所示：
 
 - MySQL sink
 
-    - TiCDC 不拆分表内事务，**保证**单表事务一致性，但**不保证**与上游表内事务的顺序一致。如果事务间的主键有交集，依然可以保证与上游顺序一致。
-    - TiCDC 以表为单位拆分跨表事务，**不保证**跨表的事务始终一致。
+    - TiCDC 不拆分单表事务，**保证**单表事务原子性。
+    - TiCDC **不保证**下游事务执行顺序和上游完全一致。
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:58]:
```suggestion
    - TiCDC 不拆分单表事务，**保证**单表事务的原子性。
    - TiCDC **不保证**下游事务的执行顺序和上游完全一致。
```

**Review by TomShawn — APPROVED** [review]
LGTM

File: ticdc/ticdc-overview.md [hunk 4]
```diff
@@ -54,8 +54,9 @@ TiCDC 的系统架构如下图所示：
 
 - MySQL sink
 
-    - TiCDC 不拆分表内事务，**保证**单表事务一致性，但**不保证**与上游表内事务的顺序一致。如果事务间的主键有交集，依然可以保证与上游顺序一致。
-    - TiCDC 以表为单位拆分跨表事务，**不保证**跨表的事务始终一致。
+    - TiCDC 不拆分单表事务，**保证**单表事务的原子性。
```
Comment (TomShawn) [inline ticdc/ticdc-overview.md:57]:
```suggestion
    - TiCDC 不拆分单表事务，**保证**单表事务的原子性。 
```

**Pull request merged**